doc/telestrat.sgml

```diff
@@ -0,0 +1,185 @@
+
+This header is used for Telemon 3.0. 
+
+Anyway, for Telemon 2.4, there is no file management, there is no TAPE routine in Telemon, there is no way to load a binary easily. 
+
+Stratsed (the Telestrat operating system) handles files management. Stratsed is loaded to memory from floppy disk. 
+
+There is no tool to insert a binary in a Stratsed floppy disk. 
+
+The only way to load a binary (for Telemon 2.4) is to :
+<itemize>
+<item>remove the 20 bytes header
+<item>download osdk : http://osdk.defence-force.org/index?page=download
+<item>use Floppybuilder in OSDK to insert the binary with the tool (please read FloppyBuilder manual to insert your binary, and to start microdisc boot sector when Telestrat starts)
+</itemize>
+
+Please note also, that the binary converted into TAP file, will not produce a right stratsed file when tap2dsk and old2mfm are used. You will be in the case that Telestrat/Stratsed crashed when you do "DIR" command.
+
+If you know the Stratsed disk format, please contact the author of this doc.
+
+
+<sect>Memory layout<p>
+
+In the standard setup, cc65-generated programs use the memory from
+&dollar;0801 to &dollar;9800; so, nearly 37K of memory (including the stack) is
+available. ROM calls are possible without further precautions.
+
+
+Special locations:
+
+<descrip>
+  <tag/Stack/
+  The C runtime stack is located at &dollar;97FF (or &dollar;B3FF), and grows
+  downwards.
+
+  <tag/Heap/
+  The C heap is located at the end of the program, and grows towards the C
+  runtime stack.
+
+</descrip><p>
+
+
+
+<sect>Platform-specific header files<p>
+
+Programs containing Telestrat -specific code may use the <tt/telestrat.h/ header file.
+
+
+<sect1>Telestrat-specific functions<p>
+
+The functions listed below are special for the Telestrat. See the <url
+url="funcref.html" name="function reference"> for declaration and usage.
+
+<itemize>
+<item>explode
+<item>ping
+<item>shoot
+<item>zap
+<item>oups
+</itemize>
+
+
+<sect1>Hardware access<p>
+
+The following pseudo variables declared in the <tt/telestrat.h/ header file do allow
+access to hardware located in the address space. Some variables are
+structures; accessing the struct fields will access the chip registers.
+
+<descrip>
+
+  <tag><tt/VIA/</tag>
+  Access to the VIA (Versatile Interface Adapter) chip is available via the
+  <tt/VIA/ variable. The structure behind this variable is explained in <tt/_6522.h/.
+
+</descrip><p>
+
+
+<sect>Loadable drivers<p>
+
+<sect1>Extended memory drivers<p>
+
+No extended memory drivers are currently available for the Telestrat.
+
+
+<sect1>Joystick drivers<p>
+
+<descrip>
+
+telemon 2.4 & 3.0 manages joysticks but it had been handled yet.
+
+</descrip>
+
+
+<sect1>Mouse drivers<p>
+
+<descrip>
+
+Telestrat manages also mouse, but it had been no handled yet in this version.
+
+</descrip>
+
+<sect1>RS232 device drivers<p>
+
+<descrip>
+
+Telestrat has a RS232 port, but it's not used
```

Comment:
For TGI, i should have a look, because telemon (the right telestrat rom) has not same calls than the atmos, not the same parameters, and as i did a lot of reverse engineer for telemon, it require some work. For joysticks, there is also routines for joystick management in telemon, i will have a loop too.